ui/raidboss/data/07-dt/ultimate/futures_rewritten.ts

```diff
@@ -249,7 +278,120 @@ const triggerSet: TriggerSet<Data> = {
       },
     },
     // P2 -- Usurper Of Frost
+    {
+      id: 'FRU P2 Quadruple Slap',
+      type: 'StartsUsing',
+      netRegex: { id: '9CFF', source: 'Usurper of Frost' },
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'FRU P2 Quadruple Slap Cleanse',
+      type: 'GainsEffect',
+      netRegex: { effectId: '1042', source: 'Usurper of Frost' },
+      condition: (data) => data.CanCleanse(),
+      infoText: (data, matches, output) =>
+        output.cleanse!({ player: data.party.member(matches.target).nick }),
+      outputStrings: {
+        cleanse: {
+          en: 'Cleanse ${player}',
+        },
+      },
+    },
+    {
+      id: 'FRU P2 Diamond Dust',
+      type: 'StartsUsing',
+      netRegex: { id: '9D05', source: 'Usurper of Frost', capture: false },
+      response: Responses.bigAoe(),
+    },
+    {
+      id: 'FRU P2 Axe/Scythe Kick Collect',
+      type: 'StartsUsing',
+      // 9D0A - Axe Kick (be out), 9D0B - Scythe Kick (be in)
+      netRegex: { id: ['9D0A', '9D0B'], source: 'Oracle\'s Reflection' },
+      // there are 2 actors 180 degrees apart, but we only need to collect one
+      run: (data, matches) => data.p2AxeScytheSafe = matches.id === '9D0A' ? 'out' : 'in',
+    },
+    {
+      id: 'FRU P2 Icicle Impact Initial Collect',
+      type: 'StartsUsing',
+      netRegex: { id: '9D06', source: 'Usurper of Frost' },
```

Comment:
Hey passerby here. I was curious as to how this was done since I was attempting something similar.

 I came across some logs where the source was NOT "Usurper of Frost". In some cases it was either "Fatebreaker" or "Fatebreaker's Image" that was first casting this icicle impact. I tried this trigger on my end with a log that had this Fatebreaker casting Icicle Impact and it was stating the wrong initial location.

Reply:
The game will reuse hidden actors, especially in longer fights like an ultimate.  If the combatant data is stale, the ACT_FFXIV plugin will emit the last name it saw, which may not be correct when an actor gets reused.

Reply:
This reuse of hidden actors is why I don't bother including a `source` property in my personal triggers anymore.